#!/bin/bash

#############################################################
# Convenience script to get the overall alignment percent 
# from all .err files in the HISAT2 (6_align.sh) log folder
# and the count of properly aligned reads (script 7a must 
# thus be run first).
#############################################################

# I/O
# align_err_dir = path to the log dir containing HISAT2 alignment .err files
# alignment_count_dir = path to the alignment count files (generated by 7a...sh)
# output = path to output stats to
align_err_dir=logs/6_align
alignment_count_dir=../results/sequencing/alignment/properly_paired_counts
output=../results/sequencing/alignment/aligned_read_counts.tsv

# Create an array of the alignment err files
all_align_err_files=$(find $align_err_dir -name *.err)

# Header for output file
echo -e "sample\toverall_alignment_percent\taligned_read_n" > $output

# Loop through each file, get the sample name and overall alignment, and output to stdout
for err in $all_align_err_files; do
	# Get the sample name
	sample=$(grep -Po "(?<=SortSam -INPUT ).*(?= -OUTPUT)" $err)
	sample=$(basename $sample | grep -Po ".*(?=\.sam)")

	# Get the overall alignment percent (output in HISAT logs)
	overall_alignment_percent=$(grep -Po "[0-9]+\.[0-9]+(?=% overall alignment rate)" $err)

	# Get the count of properly aligned reads (output from script 7a)
	alignment_count=$(cat $alignment_count_dir/$sample.alignment_count.txt)

	# Write the sample's alignment percent/count
	echo -e "$sample\t$overall_alignment_percent\t$alignment_count" >> $output
done